lib/localisation/src/preprocessing/keyence.rs

```diff
@@ -1 +1,96 @@
+use heapless::Vec;
 
+#[derive(PartialEq, Debug)]
+pub enum SensorChecks {
+    Acceptable,
+    Unnaceptable,
+}
+
+/// This struct checks if both the current and previous keyance data is in disagreement
+pub struct KeyenceAgrees {
+    previous_keyance_agreement: bool,
+}
+
+impl Default for KeyenceAgrees {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl KeyenceAgrees {
+    pub fn new() -> Self {
+        KeyenceAgrees {
+            previous_keyance_agreement: true,
+        }
+    }
+
+    pub fn check_keyence_agrees(&mut self, keyence_data: Vec<bool, 2>) -> SensorChecks {
+        if keyence_data[0] != keyence_data[1] && !self.previous_keyance_agreement {
```

Comment:
I guess keyence_data[0] != keyence_data[1] doesnt have to be repeated twice but i dont think that will make any difference lol